src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,170 @@
+      this.flags as ProcessFlags,
+      "App Upgrade",
+    );
+    const appInstallationId = await withAppInstallationId(
+      this.apiClient,
+      UpgradeApp,
+      this.flags,
+      this.args,
+      this.config,
+    );
+
+    const currentAppInstallationResponse =
+      await this.apiClient.app.getAppinstallation({
+        appInstallationId: appInstallationId,
+      });
+    const currentAppInstallation = currentAppInstallationResponse.data;
+
+    const currentAppUuid: string = currentAppInstallation.appId as string;
+    const currentApp = await getAppFromUuid(this.apiClient, currentAppUuid);
+    const currentAppVersionUuid = (
+      currentAppInstallation.appVersion as { current: string }
+    ).current;
+
+    const currentAppVersion = await getAppVersionFromUuid(
+      this.apiClient,
+      currentAppUuid,
+      currentAppVersionUuid,
+    );
+
+    const upgradeCandidates =
+      await this.apiClient.app.listUpdateCandidatesForAppversion({
+        appId: currentAppUuid,
+        baseAppVersionId: currentAppVersionUuid,
+      });
+
+    if (this.flags["show-candidates"]) {
+      process.addInfo(
+        <Text>
+          Your {currentApp.name} in Version ${currentAppVersion.externalVersion}{" "}
+          can currently be upgraded to the following Versions:
+        </Text>,
+      );
+      for (let i = 0; i < upgradeCandidates.data.length; i++) {
+        process.addInfo(
+          <Text>{upgradeCandidates.data[i].externalVersion}</Text>,
+        );
+      }
+      return;
+    }
+
+    let targetAppVersion;
+
+    if (this.flags["target-version"] == "latest") {
+      targetAppVersion =
+        await getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates(
+          this.apiClient,
+          currentAppUuid,
+          currentAppVersionUuid,
+        );
+      if (!targetAppVersion) {
+        process.complete(
+          <Success>
+            Your App is already at the lastest available or possible version. ✅
+          </Success>,
+        );
+      }
+    } else {
+      targetAppVersion = await getAvailableTargetAppVersionFromExternalVersion(
+        this.apiClient,
+        currentAppUuid,
+        currentAppVersionUuid,
+        this.flags["target-version"],
+      );
+      if (!targetAppVersion) {
+        process.error(
+          "Given target version is not a valid upgrade candidate for the current version of your app. \n" +
+            "You can check valid upgrade candidates for your apps with the flag '--show-candidates'",
+        );
+        return;
+      }
+    }
+
+    if (!targetAppVersion) {
+      process.error("Target App version could not be determined properly.");
+      return;
+    }
+
+    if (!this.flags.force) {
+      const confirmed = await process.addConfirmation(
+        <Text>
+          Confirm Upgrading {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (description here) to Version{" "}
+          {targetAppVersion.externalVersion}
```

Comment:
```suggestion
          Confirm upgrading {currentApp.name}{" "}
          {currentAppVersion.externalVersion} (description here) to version{" "}
          {targetAppVersion.externalVersion}
```

https://www.grammarly.com/blog/capitalization-rules/ 😉

Reply:
Also, what's up with `description here`?

Reply:
mentioned typos have been corrected, and then some.